src/canvas.jl

```diff
@@ -50,7 +50,9 @@ function lines!(
     y1::Number,
     x2::Number,
     y2::Number,
-    color::UserColorType,
+    c_or_v1::Union{AbstractFloat,UserColorType},  # either floating point values or colors
+    c_or_v2::Union{AbstractFloat,UserColorType} = nothing,
+    col_cb = nothing,  # callback (map values to colors)
```

Comment:
It's not immediately clear to me what's the purpose of these keywords. Can you explain it a bit?

Reply:
For the old version we only needed one color (`color_or_value1` = `c_or_v1` = `a color` (int, or tuple)). Now, since I need to interpolate the color along the whole line segments, I need to pass the raw `height` values (`floats`) before mapping them to colors (since mapping to ansi 256 color is non-bijective, non-linear, so I cannot interpolate colors by using some kind of averaging). For a line I need two values `v1` and `v2` so that along the line segments, we can compute color values using `col_cb` which calls `cmapcolor(z, ...)` from `src/colormaps.jl`. Is that clearer ? If not, I will duplicate the code to make it clearer, I just wanted to reuse the code as much as possible here.

Reply:
As for using an interpolation function from an external package I believe that this is not needed for 2 reasons: the interpolation is linear (we don't need splines, etc...) and second this is segment based so there is no `simd` computations involved, and we won't benefit from any vectorization, or whatever optimization.

Reply:
But this PR allocates more than previously and I shall for sure fix that before merging ;)

Reply:
`Are these shared context values for all lines!?` `lines!` is called for each line segment that you see on the image.

Reply:
I'll have a look tomorrow, I think we need to enhance `3D` -> `2D` transformations by doing a single matmat product before going further with this.

Reply:
This is much clearer to me! I still have a few questions that hopefully can be answered to ensure that I get your intention correctly:

- can I assume `c_or_v1` is the color for point `(y1, x1)` and `c_or_v2` is the color for point `(y2, x2)`?
- And, for the scalar indexed-color case. How does the interpolation solve/workaround the non-linear issue?
- Why we don't do the "same" interpolation for the tuple case?

Reply:
- yes, exactly a line going from `(x1, y1, c1)` to `(x2, y2, c2)`
- its not the interpolation, it's the order: invalid := interpolate(compute_colors(...)), valid := compute_colors(interpolate(...))
- which tuple case ? do you mean when we use colors as `RGB`s ?